cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/AbstractContext.java

```diff
@@ -230,6 +231,11 @@ public DynamicDatasetCache getDatasetCache() {
 
   protected <T extends Dataset> T getDataset(String namespace, String name, Map<String, String> arguments,
                                              AccessType accessType) throws DatasetInstantiationException {
+    if (NamespaceId.SYSTEM.getNamespace().equalsIgnoreCase(namespace)) {
+      throw new DatasetInstantiationException(String.format("Dataset %s cannot be instantiated from %s namespace. " +
```

Comment:
Do we need user ? No one is allowed to read/write stuff in SYSTEM NS right ? So the above message sounds more like user does not have privilege rather than dataset in SYSTEM cannot be accessed regardless who the user is it. If some users can then having user will absolutely make sense. Let me know.